Article categories

ehaivala opened the pull request:
Add categories for articles. Add category detail view which lists the articles in the category. Also some minor fixes here and there and new category factory that is used in tests and `publish_test_articles` management command.

Refs RND-9

JsseL commented on cmsplugin_articles_ai/views.py:
```diff
@@ -65,12 +67,40 @@ def get_context_data(self, **kwargs):
         return context
 
 
+class CategoryView(DetailView):
```
Couldn't you subclass this from `ArticleListView` since this should be pretty much a view with a list of articles. `ArticleListView` also provides tag filtering so you could filter a category with tags.

JsseL commented on cmsplugin_articles_ai/templates/cmsplugin_articles_ai/category_detail.html:
```diff
@@ -0,0 +1,23 @@
+            {% include "cmsplugin_articles_ai/tag_list.html" with tags=all_tags %}
+        </div>
+    </div>
+{% endblock content %}
```
Shouldn't this file be called just a generic `article_list.html` I don't see anything category specific in this.

JsseL commented on cmsplugin_articles_ai/factories.py:
```diff
@@ -47,6 +55,7 @@ class Meta:
     title = factory.Faker("catch_phrase")
     slug = factory.LazyAttribute(lambda article: slugify(article.title))
     published_from = timezone.now() - timedelta(days=1)
+    category = factory.Iterator(Category.objects.all())
```
Not sure if I like this approach since this seems to cause the need for repeating `CategoryFactory()` in multiple places which isn't nice. I would keep the existing tests as is i.e. not requiring categories since this plugin needs to work also without categories IMO (as it seems to do in this PR). But keeping the existing tests as is would make sure of that.

Some ideas of getting rid of the category requirement in this factory:
- Add some checking so that category is assigned only if categories exist
- Make another factory that assigns also the category and keep category as null in here

JsseL commented:
One more, really important thing. Add "Features" to the README where you explain at least the new category support, how it can or can not be used. Something that helps the next developer to decide whether she should try the plugin or not.

ehaivala commented:
@JsseL: I did some changes as suggested.

I changed `CategoryView` to subclass `ArticleListView` and added support for simultaneous category and tag filtering for `CategoryView`. Both views use the same template now. I also cleaned the factories and test cases so that `CategoryFactory` isn't required everywhere anymore. I moved the contents of `.coveragerc` to `setup.cfg` as well.

I added feature list to readme and converted it from Markdown to reStructuredText.
